test/mainnet-fork-test-sushiswapStakingReward-USTWETH.js

```diff
@@ -0,0 +1,280 @@
+        let strategyAsProxy;
+        let feeRewardForwarder;
+
+
+        async function setupExternalContracts() {
+          underlying = await IERC20.at(sushiLPAddress);
+          cropToken = await IERC20.at(MFC.SUSHI_ADDRESS);
+          cropPool = await IMasterChef.at(MFC.SUSHISWAP_MASTER_CHEF);
+          token0Path = tokenLiquidationPaths[0]; // [MFC.SUSHI_ADDRESS, MFC.WETH_ADDRESS, tokens[0]];
+          token1Path = tokenLiquidationPaths[1]; // [MFC.SUSHI_ADDRESS, MFC.WETH_ADDRESS, tokens[1]];
+          token0 = await IERC20.at(tokens[0]);
+          token1 = await IERC20.at(tokens[1]);
+
+          sushiswapRouter = await UniswapV2Router02.at(MFC.SUSHISWAP_ROUTER_ADDRESS);
+
+          // master chef does not have this
+          cropPool.balanceOf = async (address) => {
+            const info = await cropPool.userInfo(poolID, address);
+            return info.amount;
+          }
+          console.log("External setup finished")
+        }
+
+        async function resetBalance() {
+          // Give whale some ether to make sure the following actions are good
+          await send.ether(etherGiver, token0Whale, "1000000000000000000");
+          await send.ether(etherGiver, token1Whale, "1000000000000000000");
+
+          await token0.transfer(farmer1, await token0.balanceOf(token0Whale), {from: token0Whale});
+          await token1.transfer(farmer1, await token1.balanceOf(token1Whale), {from: token1Whale});
+          let token0Balance = await token0.balanceOf(farmer1);
+          let token1Balance = await token1.balanceOf(farmer1);
+
+          await token0.approve(sushiswapRouter.address, token0Balance, {from:farmer1});
+          await token1.approve(sushiswapRouter.address, token1Balance, {from:farmer1});
+          await sushiswapRouter.addLiquidity(
+            token0.address,
+            token1.address,
+            token0Balance, // desired
+            token1Balance, // desired
+            1,  // min
+            1,  // min
+            farmer1,
+            1760144767,
+            { from: farmer1 }
+          );
+
+          farmerBalance = await underlying.balanceOf(farmer1);
+          let farmerBalance1 = new BigNumber(farmerBalance);
+          console.log("LP provided: ", farmerBalance1.toFixed())
+        }
+
+        async function setupCoreProtocol() {
+          // deploy storage
+          storage = await Storage.new({ from: governance });
+
+          feeRewardForwarder = await FeeRewardForwarder.new(storage.address, underlying.address, MFC.UNISWAP_V2_ROUTER02_ADDRESS, { from: governance });
+          // set up controller
+          controller = await Controller.new(storage.address, feeRewardForwarder.address, {
+            from: governance,
+          });
+
+          await storage.setController(controller.address, { from: governance });
+
+          // set up the vault with 100% investment
+          vault = await makeVault(storage.address, underlying.address, 100, 100, {from: governance});
```

Comment:
why the vault has 100% investment?

Reply:
That is part of the original Sushi test file. Did not change the numbers. Was assuming this is the standard value.

Reply:
Also there is no reason to not invest 100%, as withdrawal is always possible still.